agent/app/agent_unix.go

```diff
@@ -0,0 +1,77 @@
+// +build linux
```

Comment:
I thought about it, but decided against that as there were no compelling advantages of keeping this in another package. It can be argued that this is one of the things that the agent app needs to be doing, which is why it's here. Do you have any particular reason for moving this to another package?